### PR TITLE
Backport of PR 2187 in branch 1.24.x

### PR DESCRIPTION
**Pull request by olamy**
- [JENKINS-60995] Fix downstream build link
- Update to use pipeline-build-step 2.14-rc430.69345dedbb03

# Description

See [JENKINS-60995](https://issues.jenkins-ci.org/browse/JENKINS-60995).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: acceptance-tests/src/test/resources/io/blueocean/ath/offline/pipeline/DownstreamLinkTest/sequentialStages.upstream.groovy [hunk 1]
```diff
@@ -0,0 +1,30 @@
+pipeline{
+  agent any
+  stages {
+    stage ('Sequential') {
+      stages {
+        stage("Sequential 1 with child in it"){
+          steps{
+            build "TestJob"
```
Comment (Dohbedoh) [inline acceptance-tests/src/test/resources/io/blueocean/ath/offline/pipeline/DownstreamLinkTest/sequentialStages.upstream.groovy:8]: shouldn't this be "downstreamJob" ?

File: acceptance-tests/src/test/resources/io/blueocean/ath/offline/pipeline/DownstreamLinkTest/sequentialStages.upstream.groovy [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          }
+        }
+        stage("Sequential 2 with child in it"){
+          steps{
+            build "TestJob"
```
Comment (Dohbedoh) [inline acceptance-tests/src/test/resources/io/blueocean/ath/offline/pipeline/DownstreamLinkTest/sequentialStages.upstream.groovy:13]: ditto

File: acceptance-tests/src/test/resources/io/blueocean/ath/offline/pipeline/DownstreamLinkTest/sequentialStages.upstream.groovy [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          }
+        }
+        stage("Sequential 3 with child in it using Script"){
+          steps{
+            build "TestJob"
```
Comment (Dohbedoh) [inline acceptance-tests/src/test/resources/io/blueocean/ath/offline/pipeline/DownstreamLinkTest/sequentialStages.upstream.groovy:18]: ditto

File: acceptance-tests/src/test/java/io/blueocean/ath/offline/pipeline/DownstreamLinkTest.java [hunk 4]
```diff
@@ -25,21 +29,47 @@
...
         upstreamJob.getRunDetailsPipelinePage().open(1);
+        upstreamJob.getRunDetailsPipelinePage().open(2);
+        upstreamJob.getRunDetailsPipelinePage().open(3);
 
         find("//*[contains(text(),'Triggered Builds')]").isVisible(); // Heading for table of builds
         find("//*[contains(text(),'downstreamJob')]").isVisible(); // row pointing to downstream build
     }
+
+    private String getJobScript(String name) throws IOException {
+        return new String(IOUtils.toByteArray(getClass().getResource(DownstreamLinkTest.class.getName() + "/" + testName.getMethodName() + "." + name + ".groovy")));
```
Comment (olamy) [inline acceptance-tests/src/test/java/io/blueocean/ath/offline/pipeline/DownstreamLinkTest.java:73]: @Dohbedoh the problem is here :) I'm fixing that especially as 1.24.x doesn't have same level of commons-io dependencies

**Comment by olamy** [comment]
Still waiting for a release of pipeline-build-step-plugin
I can't but did that jenkins-infra/repository-permissions-updater#2015

**Pull request merged**